src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -3108,33 +3132,109 @@ private void BindArrayInitializerExpressions(InitializerExpressionSyntax initial
                     new PointerTypeSymbol(elementType));
             }
 
+            TypeSymbol type = inLegalPosition ? GetStackAllocType(node, elementType, diagnostics) : null;
```

Comment:
Not related to this PR: I'm surprised that the `type` isn't set when `stackalloc` is bad position. I would think that would negatively affect the semantic model behavior. 
Would you mind adding a test to confirm, and if there is a problem, either fixing it or filing an issue? Thanks

Reply:
>I'm surprised that the type isn't set when stackalloc is bad position. 

in addition to that, type isn't set when stackalloc is not directly in a local declaration init.